### PR TITLE
Adding esa_hubble module

### PR DESCRIPTION
**Pull request by esdc-esac-esa-int**
Adding first version of the 'esa_hubble' module

### CODE REVIEW

**Comment by keflavich** [comment]
Thanks for the contribution.  This is a big PR.  Would it be possible for you to have a colleague at ESA do a code review here?

**Comment by keflavich** [comment]
Could you also explain how this is different from / related to ESASky?

**Comment by esdc-esac-esa-int** [comment]
> Thanks for the contribution. This is a big PR. Would it be possible for you to have a colleague at ESA do a code review here?

Hi, thanks for the reply. Sure, I will contact @jcsegovia for him to take a look at it.

**Comment by esdc-esac-esa-int** [comment]
> Could you also explain how this is different from / related to ESASky?
This module is intended for accessing the European Hubble Archive from Astroquery. This is a separated project from ESASky. The European Hubble Archive is a full-fledged Hubble Archive while ESASky is a data discovery portal for several ESA missions.

**Comment by esdc-esac-esa-int** [comment]
> > Could you also explain how this is different from / related to ESASky?

This module is intended for accessing the European Hubble Archive from Astroquery. This is a separated project from ESASky. The European Hubble Archive is a full-fledged Hubble Archive while ESASky is a data discovery portal for several ESA missions.

**Comment by jcsegovia** [comment]
@jduran I have been reviewing your code.
- When executing methods that imply data download, I think it is useful to inform the user that a file has been created.
- Please, move Conf initialization to __init__.py
- Please, change private attributes names: replace double underscores by single ones.
- Please,  check point 4 in doc because it seems it is not working.

**Comment by keflavich** [comment]
Might it be appropriate to make this a submodule of ESA, then?  So we'd have, e.g., `esa.hubble`, `esa.esasky`, `esa.gaia`, etc?

**Comment by esdc-esac-esa-int** [comment]
> Might it be appropriate to make this a submodule of ESA, then? So we'd have, e.g., `esa.hubble`, `esa.esasky`, `esa.gaia`, etc?

**Comment by esdc-esac-esa-int** [comment]
> Might it be appropriate to make this a submodule of ESA, then? So we'd have, e.g., `esa.hubble`, `esa.esasky`, `esa.gaia`, etc?

There is a decision to be made on this at ESA management level. The resolution is expected by December-January.

**Comment by keflavich** [comment]
On the name scheme, see #1314

**Comment by esdc-esac-esa-int** [comment]
Could any of you please help me with this. I cannot pass the automated validation:

urllib.error.URLError: <urlopen error An attempt was made to connect to the internet by a test that was not marked `remote_data`. The requested host was: archives.esac.esa.int

Tests I did do not access to internet

File: astroquery/esa_hubble/tests/test_esa_hubble.py [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+        parameterst['verbose'] = False
+        dummyTapHandler = DummyESAHubbleTapHandler("launch_job", parameterst)
+
+        parameters = {}
+        parameters['observation_id'] = "J6FL25S4Q"
+        parameters['calibration_level'] = "RAW"
+        parameters['verbose'] = False
+        dummyHandler = DummyHandler("get_product", parameters)
+        ehst = ESAHubbleClass(dummyHandler, dummyTapHandler)
+        ehst.get_product("J6FL25S4Q", "RAW")
```
Comment (keflavich) [inline astroquery/esa_hubble/tests/test_esa_hubble.py]: This test, and the other tests in this file, look like they require an internet connection to work.  You need to mark these with the `@remote_data` decorator.  See https://astroquery.readthedocs.io/en/latest/testing.html#test-module-remote-py

**Comment by esdc-esac-esa-int** [comment]
I don't know why I am seeing so many references to CDS module in my Travis report:
https://travis-ci.org/astropy/astroquery/jobs/480808270
Am I missing something?

**Comment by keflavich** [comment]
No, I don't know what's going on.  @bsipocz, any ideas?

You might rebase against master once @bsipocz finishes the infrastructure refactor in #1334; I suspect and hope that will solve this issue.

**Comment by esdc-esac-esa-int** [comment]
Any news on the problem?

**Comment by keflavich** [comment]
We're still working on it.  Sorry about the delay.

**Comment by keflavich** [comment]
I just tried to test this locally and discovered some of the huge data files.  We can't have 15 MB files in our test data, since the tests and test data have to be distributed with the source code.  Could you try to come up with a test that doesn't require such a large file?

**Comment by keflavich** [comment]
The PR definitely needs a rebase or at least some sort of history cleanup; I attempted to rebase and ran into a lot of conflicts in the .gitignore and some conflicts _within_ the `esa_hubble` package.

Also, astropy-helpers needs to be updated; that *may* resolve the travis-ci issues.

**Pull request closed without merging**